src/Sound/Tidal/UI.hs

```diff
@@ -1439,6 +1445,47 @@ _arp name p = arpWith f p
         thumbup xs = concatMap (\x -> [thumb,x]) $ tail xs
           where thumb = head xs
 
+{- | `rolled` plays each note of a chord quickly in order, as opposed to simultaneously; to give a chord a harp-like effect.
+This will played from the lowest note to the highest note of the chord
+@
+rolled $ n "c'maj'4" # s "superpiano"
+@
+
+You can reverse this order by using the `rolled'` function
+
+@
+rolled' $ n "c'maj9" # s "superpiano"
+@
+
+And you can use `rolledBy` or `rolledBy'` to specify the length of the roll. The value in the passed pattern
+is the divisor of the cycle length.
+
+@
+rolledBy "<1 2 4 8 16 3>" $ note "c'maj9" # s "superpiano"
+@
+-}
+
+rolledWith :: ([Event a] -> [EventF (ArcF Time) a]) -> Ratio Integer -> Pattern a -> Pattern a
+rolledWith f t = withEvents aux
+         where aux es = concatMap (steppityIn) (groupBy (\a b -> whole a == whole b) $ sortOn whole (f es))
+               steppityIn xs = mapMaybe (\(n, x) -> shiftIt n (length xs) x) $ enumerate xs
+               shiftIt n d (Event c (Just (Arc s e)) a' v) = do
+                         a'' <- subArc (Arc newS e) a'
+                         return (Event c (Just $ Arc newS e) a'' v)
+                      where newS = s + (dur * fromIntegral n)
```

Comment:
I think this will result in patterns which aren't well formed, by moving events outside the original query.
Fixing this would be a little complicated I think. The query would need to be made bigger, subtracting the roll duration from the start and adding the duration to the end of the query arc. Then after the calculation done, resulting events trimmed to the original query, and removed completely if they're outside that original query.

Reply:
Hello Alex! I'm just taking the time to try to finish this topic. My understanding is that the behavior you describe should already be implemented. For example `s "superpiano" <| n (rolledBy "1.5" "[0,1,2,3]")
`

will result in 

```
 (0>1)|n: Note {unNote = 0.0}n, s: "superpiano"
(⅜>1)|n: Note {unNote = 1.0}n, s: "superpiano"
(¾>1)|n: Note {unNote = 2.0}n, s: "superpiano"
```

So, practically, when the time arc of the pattern gets bigger (in theory), the overhanging events are truncated. 

And that's what you meant, right?

Reply:
I'm finding it hard to get my head around this in the time I have available.. 

But what I meant is something like.. Tidal calculates events in a window given by the query. This makes functions like this quite difficult, when we want to cause one event to happen after an existing one, in at least two ways. We might query a timespan that does not contain any events, but there was one happening just before that timespan that should cause one in the window we're asking for. But we don't know about that. The other problem is that we might have an event in the window but it should cause an event outside that window. From what I could tell, both are an issue here.

Reply:
I'll merge for now and will try to make a failling test at some point

Reply:
Now I play with it, it seems I was misunderstanding something, it seems to work fine. Sorry for holding this up!

Reply:
Amazing work - thanks Thomas and Alex